interfaces/builtin/upower_control.go

```diff
@@ -0,0 +1,215 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+)
+
+const upowerControlPermanentSlotAppArmor = `
+# Description: Allow operating as the UPower service.
+
+network netlink,
+network netlink raw,
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+
+dbus (send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member={Request,Release}Name
+   peer=(name=org.freedesktop.DBus),
+
+dbus (receive, send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member=GetConnectionUnixProcessID
+   peer=(label=unconfined),
+
+dbus (receive, send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member=GetConnectionUnixUser
+   peer=(label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.freedesktop.UPower",
+
+# Allow traffic to/from our path and interface with any method
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/UPower{,/**}
+    interface=org.freedesktop.UPower*,
+
+# Allow traffic to/from org.freedesktop.DBus for the UPower service
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/UPower{,/**}
+    interface=org.freedesktop.DBus.*,
+
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/login1{,/**}
+    interface=org.freedesktop.DBus.*
+    peer=(label=unconfined),
+
+# Allow access to logind service as we need to query it for possible
+# power states and trigger these when the battery gets low and the
+# system enters a critical state.
+dbus (send)
+    bus=system
+    path=/org/freedesktop/login1{,/**}
+    interface=org.freedesktop.login1.Manager
+    member={CanPowerOff,CanSuspend,CanHibernate,CanHybridSleep,PowerOff,Suspend,Hibernate,HybrisSleep},
+    peer=(label=unconfined),
```

Comment:
I will drop 'send' from the rule before and just leave this one with send. Then we have one to receive any signal from logind and one rule to allow calling just a specified set of methods. How does that sound?